wandb/sdk/launch/agent/agent.py

```diff
@@ -319,9 +315,27 @@ def finish_thread_id(
             except CommError:
                 pass
             if run_info is None:
-                _msg = "The submitted run was not successfully started"
+                # We retry for 60 seconds with an exponential backoff in case
+                # upsert run is taking a while.
+                start_time = time.time()
+                interval = 1
+                while time.time() - start_time < 60:
+                    try:
+                        run_info = self._api.get_run_info(
+                            self._entity,
+                            job_and_run_status.project,
+                            job_and_run_status.run_id,
+                        )
+                        break
```

Comment:
We can't break here unfortunately, this will always return without comm error for sweeps since the run exists but the run info will be None. We can break if run_info is not None

Reply:
Otherwise we need to sleep and try again.

Reply:
Got it